### PR TITLE
fix #214, fix List Properties

### LINKED ISSUE #214: browsing /Service.svc causes StackOverflowException
Now listening on: http://[::]:5050                                               Application started. Press Ctrl+C to shut down.                                  info: Microsoft.AspNetCore.Hosting.Internal.WebHost[1]                                 Request starting HTTP/1.1 GET http://localhost:5050/Service.svc                                                                                             Process is terminating due to StackOverflowException.
--

### PR DESCRIPTION
**Pull request by kotovaleksandr**
Fix #214 and fix wsdl properties name with DataMember attribute.

@agpreynolds Can you review my code?

### CODE REVIEW

File: src/SoapCore/MetaWCFBodyWriter.cs [hunk 1]
```diff
@@ -974,7 +979,7 @@ private bool HasBaseType(Type type)
 
 			var baseType = type.GetTypeInfo().BaseType;
 
-			return !isArrayType && !type.IsEnum && !type.IsPrimitive && baseType != null && !baseType.Name.Equals("Object");
+			return !isArrayType && !type.IsEnum && !type.IsPrimitive && baseType != null && !baseType.Name.Equals("Object") && baseType.FullName != typeof(ValueType).FullName;
```
[the remark this answers is not in the document]
Comment (agpreynolds) [reply src/SoapCore/MetaWCFBodyWriter.cs]: I think this should work and be a little bit cleaner? What type was causing the issue here?

[the remark this answers is not in the document]
Comment (kotovaleksandr) [reply src/SoapCore/MetaWCFBodyWriter.cs]: Check type DateTimeOffset: IsValueType is true and base type - System.ValueType. #214 describe how DateTimeOffset property cause StackOverflowException, recursion stay on System.Value infinity.

[the remark this answers is not in the document]
Comment (agpreynolds) [reply src/SoapCore/MetaWCFBodyWriter.cs]: So if IsValueType returns true, my suggestion would also fix the problem for DateTimeOffset. Can you think of a scenario where we would want to get properties from a ValueType?

[the remark this answers is not in the document]
Comment (kotovaleksandr) [reply src/SoapCore/MetaWCFBodyWriter.cs]: Yes, your right. I'm forget what ValueTypes (structs) cannot be inherit. Thanks!

**Pull request merged**